src/main/java/org/prebid/server/bidder/epsilon/EpsilonBidder.java

```diff
@@ -84,22 +93,37 @@ private BidRequest createOutgoingRequest(BidRequest bidRequest) {
         for (int i = 0; i < requestImps.size(); i++) {
             final Imp imp = requestImps.get(i);
             final ExtImpEpsilon impExt = parseImpExt(imp, i);
-            modifiedImps.add(modifyImp(imp, impExt));
+            final BigDecimal bidFloor = resolveBidFloor(bidRequest,
+                    imp.getBidfloorcur(),
+                    getBidFloor(imp.getBidfloor(), impExt.getBidfloor()));
+            modifiedImps.add(modifyImp(imp, impExt, bidFloor));
         }
 
         final Imp firstImp = requestImps.get(0);
         final ExtImpEpsilon extImp = parseImpExt(firstImp, 0);
         final String siteId = extImp.getSiteId();
         final Site requestSite = bidRequest.getSite();
         final App requestApp = bidRequest.getApp();
-
+        final List<String> cur = bidRequest.getCur();
+        if (cur != null && !cur.isEmpty() && !BIDDER_CURRENCY.equals(cur.get(0))) {
```

Comment:
This line at 108 and the section starting at line 120 are the same checks but written in different ways:

```java
if (cur != null && !cur.isEmpty() && !BIDDER_CURRENCY.equals(cur.get(0))) {
```
```java
            && !StringUtils.equalsIgnoreCase(bidfloorcur, BIDDER_CURRENCY)
                && StringUtils.isNotBlank(bidfloorcur)) {
```
In any case, it's OK to always overwrite bid level currency to USD without any check.  In other words, remove line 107 to 109, and just add a line somewhere in the original builder like this:

```java
 return bidRequest.toBuilder()
                .site(updateSite(requestSite, siteId))
                .app(requestSite == null ? updateApp(requestApp, siteId) : requestApp)
                .imp(modifiedImps)
                .cur(Collections.singletonList(BIDDER_CURRENCY))
                .build();
```